transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -1113,13 +1115,20 @@ struct UbufP2PCommOverlap : torch::CustomClassHolder, UbufBase {
         reduce_buf_ptr, {_tp_size, _ubufs[0].size(0), _ubufs[0].size(1)}, _ubuf.options());
       torch::sum_out(rs_output, reduce_buf, 0);
     }
+    for (int i = 0; i < _stream_compute.size(); i++) {
+      CHECK_CUDA(
+          cudaEventRecord(_stop_compute, (cudaStream_t)_stream_compute[i]));
+      CHECK_CUDA(cudaStreamWaitEvent((cudaStream_t)stream_main, _stop_compute, 0));
+    }
+    CHECK_CUDA(cudaEventRecord(_stop_send, (cudaStream_t)_stream_send));
```

Comment:
@vasunvidia So, this change is the requirement for cuda graph, right?
Given this is a functional change, it is ok for me.

Reply:
Why wasn't this synchronization needed before?

Reply:
> @vasunvidia So, this change is the requirement for cuda graph, right? Given this is a functional change, it is ok for me.

Yes. This is requirement for CUDA graph

Reply:
> Why wasn't this synchronization needed before?

Previously, _stream_recv waited on the compute streams and then stream_main waited only on _stream_recv. This is functionally correct and works for non CUDA graph case. But this causes error during CUDA graph capture since CUDA graph requires all new streams to branch out of stream_main and rejoin stream_main before end of graph capture.